packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -1,69 +1,76 @@
 import { useThree } from '@react-three/fiber';
 
 import { useVisCanvasContext } from '../vis/shared/VisCanvasProvider';
-import RatioSelectionRect from './RatioSelectionRect';
 import SelectionRect from './SelectionRect';
 import SelectionTool from './SelectionTool';
-import { useMoveCameraTo } from './hooks';
-import type { CommonInteractionProps, Selection } from './models';
-import { getEnclosedRectangle, getRatioRespectingRectangle } from './utils';
+import Box from './box';
+import { useZoomOnBox } from './hooks';
+import type { CommonInteractionProps, Rect3, Selection } from './models';
 
 type Props = CommonInteractionProps;
 
 function SelectToZoom(props: Props) {
-  const { canvasSize, canvasRatio, visRatio, dataToWorld } =
+  const { canvasSize, canvasRatio, visRatio, worldToData } =
     useVisCanvasContext();
 
   const { width, height } = canvasSize;
   const keepRatio = visRatio !== undefined;
 
   const camera = useThree((state) => state.camera);
-  const moveCameraTo = useMoveCameraTo();
+  const zoomOnBox = useZoomOnBox();
 
-  function onSelectionEnd(selection: Selection) {
-    const [worldStart, worldEnd] = getRatioRespectingRectangle(
-      ...selection.data,
-      keepRatio ? canvasRatio : undefined
-    ).map(dataToWorld); // work in world coordinates as we need to act on the world camera
+  function computeZoomBox(worldSelection: Rect3): Box {
+    const { position, scale } = camera;
+    const zoomBox = Box.fromPoints(...worldSelection);
 
-    if (worldStart.x === worldEnd.x || worldStart.y === worldEnd.y) {
-      return;
+    if (!keepRatio) {
+      return zoomBox;
     }
 
-    const zoomRect = getEnclosedRectangle(worldStart, worldEnd);
-    const { center: zoomRectCenter } = zoomRect;
+    const fovBox = Box.empty(position).expandBySize(
+      width * scale.x,
+      height * scale.y
+    );
```

Comment:
We could store this box in the context if needed

Reply:
Well, it depends on the camera scale, so it would re-render consumers of the context on every wheel/selection zoom. I'm considering more adding a camera-based utility similar to `getWorldFOV`, like `getFovSize(camera, context)`.